packages/instrumentation-openai/test/fixtures.test.js

```diff
@@ -0,0 +1,1354 @@
+    usingNock = true;
+    // https://github.com/nock/nock#modes
+    process.env.TEST_NOCK_BACK_MODE = 'lockdown';
+    // OPENAI_API_KEY needs to be set to something to avoid OpenAI
+    // constructor error. However, because of mocking, it isn't used.
+    process.env.OPENAI_API_KEY = 'notused';
+    process.env.TEST_MODEL_TOOLS = UNIT_TEST_MODEL_TOOLS;
+    process.env.TEST_MODEL_EMBEDDINGS = UNIT_TEST_MODEL_EMBEDDINGS;
+    targetService = 'openai';
+    break;
+
+  case 'regenerate-recordings':
+    // Regenerate the recorded OpenAI API responses use by "unit" test mode.
+    if (process.env.OPENAI_BASE_URL) {
+      throw new Error(
+        'OPENAI_BASE_URL is set. To regenerate-recordings, it must be empty so that recorded responses are from api.openai.com'
+      );
+    }
+    if (!process.env.OPENAI_API_KEY) {
+      throw new Error(
+        'OPENAI_API_KEY is not set. To regenerate-recordings, it must be set. Set it in your environment, or use TEST_FIXTURES_ENV_FILE="./openai.env" (see "openai.env.template").'
+      );
+    }
+    usingNock = true;
+    process.env.TEST_NOCK_BACK_MODE = 'update';
+    process.env.TEST_MODEL_TOOLS = UNIT_TEST_MODEL_TOOLS;
+    process.env.TEST_MODEL_EMBEDDINGS = UNIT_TEST_MODEL_EMBEDDINGS;
+    targetService = 'openai';
+    break;
+
+  case 'integration':
+    // Guess the target service based on the config in the environment.
+    // This is necessary because there some small differences between Azure
+    // OpenAI, OpenAI, and Ollama that test assertions need to cope with.
+    if (
+      process.env.OPENAI_BASE_URL &&
+      new URL(process.env.OPENAI_BASE_URL).port === '11434'
+    ) {
+      targetService = 'ollama';
+    } else if (process.env.AZURE_OPENAI_API_KEY) {
+      targetService = 'azure';
+    } else {
+      targetService = 'openai';
+    }
+    break;
+
+  default:
+    throw new Error('unknown test mode: ' + testMode);
+}
+
+// ---- helper functions
+
+function isPositiveInteger(val) {
+  return Number.isInteger(val) && val > 0;
+}
+
+function isExpectedServerAddress(val) {
+  if (targetService === 'openai') {
+    return val === 'api.openai.com';
+  } else if (targetService === 'azure') {
+    return val === new URL(process.env.AZURE_OPENAI_ENDPOINT).hostname;
+  } else if (targetService === 'ollama') {
+    return val === '127.0.0.1';
+  } else {
+    return false;
+  }
+}
+
+function isExpectedServerPort(val) {
+  if (targetService === 'openai') {
+    return val === 443;
+  } else if (targetService === 'azure') {
+    let port = new URL(process.env.AZURE_OPENAI_ENDPOINT).port;
+    port = port ? Number(port) : 443;
+    return val == port;
+  } else if (targetService === 'ollama') {
+    return val === Number(new URL(process.env.OPENAI_BASE_URL).port);
+  } else {
+    return false;
+  }
+}
+
+function isExpectedResponseModel(unitTestVal, requestModel) {
+  return val => {
+    if (isUnit) {
+      return val === unitTestVal;
+    } else if (targetService === 'openai') {
```

Comment:
could we generalize this to startsWith unless AZURE_OPENAI_ENDPOINT is set?

Reply:
If there is one exception (azure) in three (openai, azure, ollama), then why bother grouping the two?

Reply:
> If there is one exception (azure) in three (openai, azure, ollama), then why bother grouping the two?

mainly there aren't only three, there are many things like ollama (llama.cpp, vllm, lm studio, etc) and also many openai compatible services. I prefer to not have a constant about ollama, when prefix will pass on it and likely that makes this work with at least one other non-azure thing without a backend registry

https://github.com/open-telemetry/semantic-conventions/pull/1655

Reply:
another way to get out of this if vs case/switch could be to just assert that the attribute value is the same as the value in the completion response. We don't need to necessarily enforce a pattern on this, as it just needs to be consistent with the data returned to the user, right?

Reply:
> could we generalize this to startsWith unless AZURE_OPENAI_ENDPOINT is set?

commit 9ff242f